### DISCUSSION TITLE
Hook for simplifying usage of tv in typescript

### DISCUSSION THREAD

**Discussion by ghost** [heroui-inc/tailwind-variants#108] [Ideas]
Hi

I wanted to share with you a small hook I wrote to simplify the usage of tv in typescript. This is for slots but would be nice to have something similar built-in. 

The hook useTv.tsx looks like this:

```javascript
import { ClassProp } from "tailwind-variants";

const useTv = <T extends Record<string, (K?: ClassProp) => string>>(tv: T) => ({
  c: (slotName: keyof T, additionalClass?: string) => {
    return !additionalClass
      ? tv[slotName]()
      : tv[slotName]({ class: additionalClass });
  }
});
export default useTv;


```
That way, when declaring the component there is no need to extend the type of it or write again the name of every slot destructuring the result of tv().

```javascript
import React, { useState } from "react";
import "./styles.css";
import { tv } from "tailwind-variants";
import useTv from "./hooks/useTv";

const accordionSlots = tv({
  slots: {
    h1: "bg-red-300",
    h2: "bg-red-500 ",
    button: "bg-gray-300 mt-5 border-2 rounded-md border-black p-2"
  },
  variants: {
    isVariant: {
      true: {
        h1: "bg-blue-300 underline",
        h2: "bg-blue-500 underline"
      }
    }
  }
});

export default function App() {
  const [isVariant, setIsVariant] = useState<boolean>(false);
  const variants = { isVariant };
  const { c } = useTv(accordionSlots(variants));

  return (
    <div>
      <h1 className={c("h1")}>React + TypeScript + TailwindCSS</h1>
      <h1 className={c("h2")}>React + TypeScript + TailwindCSS</h1>

      <button className={c("button")} onClick={() => setIsVariant(!isVariant)}>
        Change variant
      </button>
    </div>
  );
}


```
You can play with it here:
https://codesandbox.io/s/talwind-variants-for-react-typescript-gmp7p9

**Comment by mskelton** [comment]
The value here seems very marginal, so definitely not something I forsee adding to the core of tailwind-variants. If it works for you great, but again doesn't add much value IMO.

**Comment by boar-is** [comment]
This should not be a "hook" since it doesn't use any React built-in hook. It would be misleading when using in RSCs i.e.

**Comment by carlbarrdahl** [comment]
Here's what I do in my projects:
- https://github.com/nextui-org/tailwind-variants/discussions/112
